### ISSUE TITLE
Support non-acct:account@domain URIs

### ISSUE THREAD

**Issue by evanminto** [Gargron/goldfinger#3]
As part of [ActivityPub support for Mastodon](https://github.com/tootsuite/mastodon/issues/1557), I need to be able to look users up by the URI of their ActivityStreams 2.0 representation (this isn't strictly necessary, but it's the best way to ensure backwards-compatibility as we move to AP). We should still use GoldFinger for this, but it seems like it only supports `acct:account@domain`-style URIs at the moment. It would be great if it could accept arbitrary URIs!

**Comment by Gargron** [comment]
No, this library does not care what you pass into webfinger. https://github.com/tootsuite/mastodon/blob/master/app/services/follow_remote_account_service.rb#L24

**Comment by evanminto** [comment]
Hm, maybe I misunderstood then, sorry! It seems like it relies on being able to split account and domain by `@` in order to find the domain to search for `.well-known`, based on some of the code I was reading here: https://github.com/tootsuite/goldfinger/blob/master/lib/goldfinger/client.rb#L78

I was also getting errors when trying to pass non-`account@domain` URIs to Goldfinger from Mastodon's end, but again, it's possible I missed something.

**Comment by redtachyons** [comment]
Hi, @evanminto Can you please provide an example uri which fails in current implementation?
